core/src/main/java/hudson/Util.java

```diff
@@ -1187,7 +1187,21 @@ public static void copyFile(@Nonnull File src, @Nonnull File dst) throws BuildEx
         cp.setTofile(dst);
         cp.setFile(src);
         cp.setOverwrite(true);
+        cp.setPreserveLastModified(true);
         cp.execute();
+        try {
+            copyFilePosixPermissions(src, dst);
+        } catch (IOException e) {
+            throw new BuildException("Cannot copy file permissions from " + src + " to " + dst, e);
```

Comment:
This function will fail on Windows now because the PosixFilePermissions-related APIs throw UnsupportedOperationException. You probably want to use `Functions#isWindows` or something to avoid this code path in that case.